### PR TITLE
Enabling TravisCI with SonarCloud for Novoalign

### PR DESCRIPTION
**Pull request by morgantaschuk**

### CODE REVIEW

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 1]
```diff
@@ -211,20 +211,10 @@ public void buildWorkflow() {
...
-//		command.addArgument("SORT_ORDER=coordinate VALIDATION_STRINGENCY=SILENT CREATE_INDEX=true");
-//		command.addArgument("TMP_DIR=" + getProperty("tmp_dir"));
-//	      	jobBam.addParent(job_novo);
-//	      	jobBam.setMaxMemory((Integer.valueOf(getProperty("picard_memory")) *2)+"");
           }
         
 
     }
+
+//FIXME
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)

**Comment by sqwci** [comment]
SonarQube analysis reported 21 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 12 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [GenomicAlignmentNovoalignWorkflow.java#L90](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L90): Define a constant instead of duplicating this literal "data_dir" 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [GenomicAlignmentNovoalignWorkflow.java#L99](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L99): Define a constant instead of duplicating this literal "/bin/" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [GenomicAlignmentNovoalignWorkflow.java#L121](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L121): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [GenomicAlignmentNovoalignWorkflow.java#L41](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L41): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [GenomicAlignmentNovoalignWorkflow.java#L123](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L123): Rename "lanes" which hides the field declared at line 49. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)
1. ![MAJOR][MAJOR] [GenomicAlignmentNovoalignWorkflow.java#L123](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L123): Rename "barcodes" which hides the field declared at line 50. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)
1. ![MAJOR][MAJOR] [GenomicAlignmentNovoalignWorkflow.java#L129](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L129): Put single-quotes around '/' to use the faster "lastIndexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MAJOR][MAJOR] [GenomicAlignmentNovoalignWorkflow.java#L130](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L130): Put single-quotes around '/' to use the faster "lastIndexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MINOR][MINOR] [GenomicAlignmentNovoalignWorkflow.java#L45](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L45): Rename this field "fastq_inputs_end_1" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)
1. ![MINOR][MINOR] [GenomicAlignmentNovoalignWorkflow.java#L46](https://github.com/oicr-gsi/novoalign/blob/04d86c70c7ab7f5ff22f417504f70198a098fdf5/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L46): Rename this field "fastq_inputs_end_2" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 2]
```diff
@@ -120,14 +125,15 @@ private Job doTrim(String read1Path, String read2Path, String read1TrimmedPath,
     @Override
     public void buildWorkflow() {
 	Workflow workflow = this.getWorkflow();
-        String[] srn=getProperty("sequencer_run_name").split(","), lanes=getProperty("lane").split(","), barcodes=getProperty("barcode").split(",");
-        for (int i = 0; i < this.fastq_inputs_end_1.length; i++) {
+        final String WORKFLOW_BIN=getWorkflowBaseDir() + File.separator+BIN_DIR+File.separator;
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 3]
```diff
@@ -120,14 +125,15 @@ private Job doTrim(String read1Path, String read2Path, String read1TrimmedPath,
...
+        String[] srn=getProperty("sequencer_run_name").split(","), llanes=getProperty("lane").split(","), lbarcodes=getProperty("barcode").split(",");
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MINOR Declare "llanes" on a separate line. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1659)
MINOR Declare "lbarcodes" on a separate line. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1659)

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 4]
```diff
@@ -38,12 +39,15 @@
...
+    private static final String BIN_DIR="bin";
+    
     private String dataDir;
     private String queue;
     private int runEnds;
-    private SqwFile [] fastq_inputs_end_1;
-    private SqwFile [] fastq_inputs_end_2;
+    private SqwFile [] fastqInputsEnd1;
+    private SqwFile [] fastqInputsEnd2;
     private String [] iusAccessions;
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MINOR Remove the "iusAccessions" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 5]
```diff
@@ -151,7 +157,7 @@ public void buildWorkflow() {
 
         	
 		//The correct format for this is using all underscores, NOT hypens
-		String runlanebarcode=srn[i]+"_"+lanes[i]+"_"+barcodes[i];
+		String runlanebarcode=srn[i]+"_"+llanes[i]+"_"+lbarcodes[i];
 		String readgroup = "-o SAM $'@RG\\tID:"+runlanebarcode+"\\tPU:"+runlanebarcode+"\\tLB:"+getProperty("rg_library")+"\\tSM:"+getProperty("rg_sample_name")+"\\tPL:"+getProperty("rg_platform")+"'";
 
 		Job job_novo=null;
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 6]
```diff
@@ -38,12 +39,15 @@
...
+    
     private String dataDir;
     private String queue;
     private int runEnds;
-    private SqwFile [] fastq_inputs_end_1;
-    private SqwFile [] fastq_inputs_end_2;
+    private SqwFile [] fastqInputsEnd1;
+    private SqwFile [] fastqInputsEnd2;
     private String [] iusAccessions;
     private String [] sequencerRunNames;
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MINOR Remove the "sequencerRunNames" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 7]
```diff
@@ -38,12 +39,15 @@
...
     private String [] lanes;
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: MINOR Remove the "lanes" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java [hunk 8]
```diff
@@ -120,14 +125,15 @@ private Job doTrim(String read1Path, String read2Path, String read1TrimmedPath,
     @Override
     public void buildWorkflow() {
```
Comment (sqwci) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

**Comment by sqwci** [comment]
SonarQube analysis reported 12 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 11 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [GenomicAlignmentNovoalignWorkflow.java#L54](https://github.com/oicr-gsi/novoalign/blob/c0d1d7538eb312a0207fab700aaec2e1e8c11034/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L54): Remove the "barcodes" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)
1. ![MINOR][MINOR] [GenomicAlignmentNovoalignWorkflow.java#L209](https://github.com/oicr-gsi/novoalign/blob/c0d1d7538eb312a0207fab700aaec2e1e8c11034/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L209): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [GenomicAlignmentNovoalignWorkflow.java#L210](https://github.com/oicr-gsi/novoalign/blob/c0d1d7538eb312a0207fab700aaec2e1e8c11034/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L210): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [GenomicAlignmentNovoalignWorkflow.java#L211](https://github.com/oicr-gsi/novoalign/blob/c0d1d7538eb312a0207fab700aaec2e1e8c11034/workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java#L211): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by morgantaschuk** [comment]
ok to test

File: workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java (hunk 8 above)
Comment (morgantaschuk) [inline workflow-novoalign/src/main/java/ca/on/oicr/pde/workflows/GenomicAlignmentNovoalignWorkflow.java]: Not going to fix this.

**Comment by morgantaschuk** [comment]
retest this please

**Pull request closed without merging**